### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by srghma**

### CODE REVIEW

**Comment by benjie** [comment]
Hi @srghma, thanks for your interest in PostGraphile!

The `/usr/bin/env` changes seem very sensible, I'd be happy to merge these.

Please can you explain why you changed the envvar from `TEST_DATABASE_URL` to `TEST_PG_URL`? You say it's "not used", but I use it almost every day...

**Comment by srghma** [comment]
@benjie tests didn't work for me with ./run-pg. https://github.com/graphile/postgraphile/blob/45e7a85d65c68d272a788d16cab61aeb296d1340/src/__tests__/utils/pgPool.ts#L4

Maybe you use postgres on localhost?

**Comment by benjie** [comment]
I use localhost PG, yes. The documentation you edited was for the Graphile Engine tests (which use `TEST_DATABASE_URL`), PostGraphile does indeed use it's own database for testing (which you can override with `TEST_PG_URL`).

**Comment by benjie** [comment]
(The Graphile Engine tests are the main ones you want to worry about, they cover a lot more cases than the PostGraphile ones. The PostGraphile ones are just a legacy to ensure compatibility is not broken - I normally only run them in CI.)

**Comment by srghma** [comment]
@benjie 

```
$ export TEST_PG_URL="postgres://postgres:password@localhost:2222/postgres"
$ npm run test
Test Suites: 7 passed, 7 total
Tests:       425 passed, 425 total
Snapshots:   38 passed, 38 total
Time:        12.772s
Ran all test suites.
```

```
$ export TEST_DATABASE_URL="postgres://postgres:password@localhost:2222/postgres"
$ npm run test
Test Suites: 4 failed, 3 passed, 7 total
Tests:       23 failed, 402 passed, 425 total
Snapshots:   16 passed, 16 total
Time:        2.833s, estimated 5s
```


why not use TEST_PG_URL everywhere?

**Comment by benjie** [comment]
@srghma Because they both reset the database but install slightly different fixtures. I often work with the Graphile Engine testing database when writing code or trying to replicate issues in PostGraphile, I don't want having recently ran a tests in PostGraphile to cause me confusing issues when I run against my Graphile Engine database.

TL;DR: because it messes with my development flow.

**Comment by srghma** [comment]
have no more questions, I undid last commit

**Comment by benjie** [comment]
Awesome, thanks @srghma! In future, it's better to open one PR per feature/fix rather than including multiple changes in one - it enables quicker merging of the non-controversial changes, and makes code review a lot easier since each change is very focussed. Further, we squash PRs into single commits, so it's nice if the final commit changes only one thing as it makes the commit message easier to understand.
